src/EditorFeatures/TestUtilities/NavigateTo/AbstractNavigateToTests.cs

```diff
@@ -306,23 +266,31 @@ public DocumentId GetActiveDocument()
 
         public ImmutableArray<DocumentId> GetVisibleDocuments()
             => ImmutableArray.Create(_workspace.CurrentSolution.Projects.First().DocumentIds.First());
+    }
 
-        [ExportWorkspaceServiceFactory(typeof(IDocumentTrackingService), ServiceLayer.Host)]
-        [Shared]
-        [PartNotDiscoverable]
-        public sealed class Factory : IWorkspaceServiceFactory
+    [Export]
+    [ExportWorkspaceServiceFactory(typeof(IDocumentTrackingService), ServiceLayer.Host)]
```

Comment:
are two exports needed?

Reply:
➡️ Discussion in https://github.com/dotnet/roslyn/pull/28957#discussion_r206579946